boshio/http_client.go

```diff
@@ -8,10 +8,30 @@ import (
 	"time"
 )
 
+func NewHTTPClient(host string, wait time.Duration) HTTPClient {
+	return HTTPClient{
+		Host: host,
+		Wait: wait,
+		Client: &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+
+				Dial: (&net.Dialer{
+					Timeout:   30 * time.Second,
+					KeepAlive: 0, // don't send keepalive TCP messages
+				}).Dial,
+
+				TLSHandshakeTimeout: 60 * time.Second,
+				DisableKeepAlives:   true,
+			},
+		},
+	}
+}
+
 type HTTPClient struct {
 	Host   string
 	Wait   time.Duration
-	Client *http.Client
+	Client httpClient
```

Comment:
Yeah, reusing that fake doesn't make this easier. Hence why I hand constructed that other type to keep it all together in just the test. I'd prefer to roll back to that so I don't have to share this interface or make the fake any larger than it needs to be (by adding something like roundtrip).

other than that, I am happy with the rest of the changes you've already made.